public/app/features/alerting/AlertRuleList.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import PageHeader from 'app/core/components/PageHeader/PageHeader';
+import AlertRuleItem from './AlertRuleItem';
+import appEvents from 'app/core/app_events';
+import { updateLocation } from 'app/core/actions';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { NavModel, StoreState, AlertRule } from 'app/types';
+import { getAlertRulesAsync, setSearchQuery } from './state/actions';
+import { getAlertRuleItems, getSearchQuery } from './state/selectors';
+
+export interface Props {
+  navModel: NavModel;
+  alertRules: AlertRule[];
+  updateLocation: typeof updateLocation;
+  getAlertRulesAsync: typeof getAlertRulesAsync;
+  setSearchQuery: typeof setSearchQuery;
+  stateFilter: string;
+  search: string;
+}
+
+export class AlertRuleList extends PureComponent<Props, any> {
```

Comment:
To me this container/page has to much logic. Would like to see usage of more "dumb components". Then this container will be AlertRulesPage or similar with alert rule list as component which in turn has alert rule items componetns.

Reply:
really? it has almost no logic (two if cases)? what logic do you mean? It is 99% a dumb page that shows AlertRuleItem items in a list, simple search filter & state filter. Think breaking up more would just make the page harder to grasp & edit. Can you describe how you would break this page up more?

Reply:
I guess the search and filter select coud be extracted to it's own component? would that make this page easier to maintain & edit?

Reply:
well that is the core responsibility of the page, to handle search & state filter and update location & rules list (through redux actions) accordingly. The responsibility of rendering each item is in the other component AlertRuleItem. 

What responsibility would you like to move out? Should the state filter drop-down be its own component and update redux state itself? It would obfuscate things I think.  Hm.. or maybe not. Maybe if both search and state dropdown was extracted to a AlertRuleListFilters component it would make it easier to reason about.

Reply:
> Maybe if both search and state dropdown was extracted to a AlertRuleListFilters component it would make it easier to reason about.

Yes that's what I meant :+1: 

> to handle search & state filter and update location & rules list

With an AlertRuleListFilters component the AlertRuleList would only need to listen for changes from that component. It doesn't matter if the AlertRuleListFilters uses textboxes, input fields, drop downs, checkboxes etc as long as it notifies parent component that filters has changed in a format agreed upon. 

As long as contract between parent component and AlertRuleListFilters is not broken the AlertRuleListFilters can be completely rewritten without any changes needed in parent component.

This is my general understanding of why it's a good idea to use dumb components.

As a side note. If the alert rule list (without filters and such) would be used in multiple places it would make sense to have that as a separate component. Then the markup `<ol className="alert-rule-list">` would be part of that component.

Reply:
I tried to reduce the amount of new components in this PR due to the refactoring being mobx -> redux. I'm all for small contained components but I think we can do these in the future?